File: src/main/java/tech/bobliu/assignment07/controller/ChatServlet.java
package tech.bobliu.assignment07.controller;

import jakarta.servlet.annotation.WebServlet;
import jakarta.servlet.http.HttpServlet;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;

import java.io.IOException;

@WebServlet(name = "chatServlet", value = {"/chat"})
public class ChatServlet extends HttpServlet {
    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
        try {
            request.getRequestDispatcher("/WEB-INF/chat.jsp").forward(request, response);
        } catch (Exception e) {
            throw new IOException(e);
        }
    }
}

File: src/main/kotlin/tech/bobliu/assignment07/controller/ChatServlet.kt
package tech.bobliu.assignment07.controller

import jakarta.servlet.annotation.WebServlet
import jakarta.servlet.http.HttpServlet
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse

@WebServlet(name = "chatServlet", value = ["/chat"])
class ChatServlet : HttpServlet() {
    override fun doGet(request: HttpServletRequest, response: HttpServletResponse) {
        request.getRequestDispatcher("/WEB-INF/chat.jsp").forward(request, response)
    }
}